### PR TITLE
Jest-tests

### PR DESCRIPTION
**Pull request by achraftha**
Change directory, 
Change Architecture for Artifact Type,
Setup Test and create first test for Artifact type

### CODE REVIEW

**Review by achraftha — COMMENTED** [review]
Change directory,
Change Artifact Type architecture,
Setup jest

File: src/listener/def_type.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { newDefType } from "../model/def_type.js";
+import { newDefType } from "../model/def_type.js.js";
```
[the remark this answers is not in the document]
Comment (achraftha) [reply src/listener/def_type.js:1]: This was an error created when I changed the directory of the files

**Review by achraftha — COMMENTED** [review]
Update imports to correct import errors

File: tests/unit/ArtifactTypeListener.spec.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      //             name: "id",
+      //             type: {
+      //                constraints: null,
+      //                description: "Unique identifier of the artifact",
+      //                type: "string",
+      //             },
+      //          },
+      //       },
+      //    },
+      // },
```
Comment (Mari0nV) [inline tests/unit/ArtifactTypeListener.spec.js:41]: What is the purpose of this comment? Maybe it should be deleted

File: src/listener/ArtifactTypeListener.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { ToscaArtifactType } from "../model/ToscaArtifactType.js";
```
[the remark this answers is not in the document]
Comment (achraftha) [reply src/listener/ArtifactTypeListener.js:1]: I'm in the process of uniformising the code to be all camelcase.

File: tests/unit/ArtifactTypeListener.spec.js (hunk 2 above)
Comment (achraftha) [reply tests/unit/ArtifactTypeListener.spec.js:41]: It's a draft i prepared for a more complex example of the test. I will delete it for now.

File: src/model/tosca_type.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+         str += this.description;
+      }
+      if (this.metadata) {
+         str += this.metadata;
+      }
+      return str;
+   }
+   static isValid(input, source) {
+      if (typeof input.derived_from != "string") {
+         // TODO
```
[the remark this answers is not in the document]
Comment (achraftha) [reply src/model/tosca_type.js:65]: I will look into it.

**Review by achraftha — COMMENTED** [review]
I updated the code with regard to the different conversations

File: src/listener/ArtifactTypeListener.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+         properties: properties,
+         mime_type: parsed_rule.value.mime_type?.value,
+         file_ext: fileExt,
+      };
+      return formattedParsedRule;
+   },
+
+   // This function is called to check if the parsed rule is correct
+   checkParsedRule(parsed_rule) {
+      //TODO: Add valid extensions to check file_ext
```
Comment (Mari0nV) [inline src/listener/ArtifactTypeListener.js:46]: should we add an issue on this repo to handle that?

Comment (achraftha) [reply src/listener/ArtifactTypeListener.js:46]: I think we will need to yes, the valid extensions are not well defined in the documentation of tosca. I think Xavier has a better idea on the issue.

File: jest.config.mjs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+   // Needed in JS codebases too because of feature flags
+   coveragePathIgnorePatterns: ["/node_modules/", ".d.ts$"],
+   testMatch: ["<rootDir>/tests/unit/**/*.spec.js"],
+   moduleFileExtensions: ["js"],
+   moduleNameMapper: {
+      "^src/(.*)$": "<rootDir>/src/$1",
+   },
+   // transform: {
+   //   ".*\\.js$": "babel-jest",
+   // },
```
Comment (Mari0nV) [inline jest.config.mjs:21]: what is the purpose of this comment?

File: jest.config.mjs [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+   transformIgnorePatterns: ["<rootDir>/node_modules/"],
+   // testResultsProcessor: "jest-sonar-reporter",
```
Comment (Mari0nV) [inline jest.config.mjs:23]: and this one?

File: src/listener/ArtifactTypeListener.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+         parsed_rule.value[key].tosca?.setId(
+            key,
+            parsed_rule,
+            "artifact_types"
+         );
+      }
+   },
+
+   // This function is called when the parser encounters a rule
+
```
Comment (Mari0nV) [inline src/listener/ArtifactTypeListener.js:15]: there is an extra empty line here

File: jest.config.mjs (hunk 7 above)
Comment (achraftha) [reply jest.config.mjs:23]: Vincent told me that he will setup sonar for this repo and once that's done, I can uncomment this line?

File: src/listener/ArtifactTypeListener.js (hunk 5 above)
Comment (Mari0nV) [reply src/listener/ArtifactTypeListener.js:46]: ok thanks

File: jest.config.mjs (hunk 7 above)
Comment (Mari0nV) [reply jest.config.mjs:23]: ok!

File: jest.config.mjs (hunk 6 above)
Comment (Mari0nV) [reply jest.config.mjs:21]: ok! let's keep it that way for now

**Review by Mari0nV — APPROVED** [review]
It looks good for me in terms of code quality.
I did not test the changes so another review from someone who can test it could be nice :)

**Pull request closed without merging**